### PR TITLE
py-currentscape: deploy and generate module

### PR DESCRIPTION
**Pull request by AurelienJaquier**
Hello everyone,

I have created a small module, currentscape, that allows scientists to easily produce currentscape figures, as in [this paper](https://elifesciences.org/articles/42722), and I would like to add it to the bluebrain modules so that it is easily accessible to scientists.
It my first pull request on github, so I hope i didn't do anything wrong.

Thank you,
Aurélien Jaquier

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-currentscape/package.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+# replace all 'x-y' with 'xY' (e.g. 'Py-morph-tool' -> 'PyMorphTool')
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-currentscape/package.py]:
```suggestion
```

File: var/spack/repos/builtin/packages/py-currentscape/package.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+class PyCurrentscape(PythonPackage):
+    """Module to easily plot currentscape."""
+
+    homepage = "https://bbpcode.epfl.ch/code/#/admin/projects/cells/currentscape"
+    url = "ssh://bbpcode.epfl.ch/cells/currentscape"
+    git = "ssh://bbpcode.epfl.ch/cells/currentscape"
+
+    version("develop", branch="master")
+    version("0.0.1.dev0", tag="currentscape-v0.0.1.dev0", preferred=True)
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-currentscape/package.py]:
```suggestion
    version("0.0.1.dev0", tag="currentscape-v0.0.1.dev0")
```
`preferred` is only needed when tagging an older version to be the default one.

File: var/spack/repos/builtin/packages/py-currentscape/package.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# Copyright 2013-2018 Lawrence Livermore National Security, LLC and other
```
Comment (matz-e) [inline var/spack/repos/builtin/packages/py-currentscape/package.py]:
```suggestion
# Copyright 2013-2020 Lawrence Livermore National Security, LLC and other
```
Travis will complain about the wrong year.

**Review by matz-e — COMMENTED** [review]
Overall, nice first PR!

Did you request to get added to the GH BlueBrain organization? I don't see a JIRA ticket for that, you should open one on the "Help Desk" with core services. That should then trigger our CI automatically.

Some small comments, otherwise looks nice!

**Comment by matz-e** [comment]
Retest this please

Seems like Jenkins is not getting triggered.

File: deploy/packages/bbp-packages.yaml [hunk 4]
```diff
@@ -128,6 +128,7 @@ packages:
       - py-bluepy@0.14.14
       - py-bluepymm@0.7.49 ^neuron%intel@19.0.4
       - py-bluepyopt@1.9.27 ^neuron%intel@19.0.4
+      - py-currentscape@0.0.1
```
Comment (matz-e) [inline deploy/packages/bbp-packages.yaml]: It's listed as `0.0.1.dev0` below…

Comment (AurelienJaquier) [reply deploy/packages/bbp-packages.yaml]: Hello!
I've been added to the GH BlueBrain organization, and I changed the version to 0.0.1, but the BlueBrain Deployment Test Build does not seem to launch. Is there something I'm missing?

**Comment by pramodk** [comment]
retest this please

**Comment by AurelienJaquier** [comment]
Can I get some help on the BLueBrain5 Deployment Test Build failing? I don't understand the error and my search engine is not very helpful...
Also, am I allowed to relaunch a test? I've read that this FetchError can happen sometime and that just relaunching the test can solve it.

**Comment by matz-e** [comment]
The phrase "Retest this please" should trigger another run. BTW, I looked at the project in Gerrit, and don't see any tags defined. Maybe this could be the issue for the fetch failure?

**Comment by AurelienJaquier** [comment]
Thank you. I added a tag in my project, I hope this will solve the problem.
Retest this please

**Comment by matz-e** [comment]
```
==> Executing phase: 'build'
==> [2020-10-22-13:52:34.043609] '/gpfs/bbp.cscs.ch/ssd/apps/hpc/jenkins/deploy/external-libraries/2020-02-01/linux-rhel7-x86_64/gcc-8.3.0/python-3.7.4-tfxecymrkn/bin/python3.7' '-s' 'setup.py' '--no-user-cfg' 'build'
setup.py:3: DeprecationWarning: the imp module is deprecated in favour of importlib; see the module's documentation for alternative uses
import imp
Traceback (most recent call last):
File "setup.py", line 12, in <module>
with open("README.rst", encoding="utf-8") as f:
FileNotFoundError: [Errno 2] No such file or directory: 'README.rst'
```

**Comment by AurelienJaquier** [comment]
Thank you Matthias for your help. I modified my files on bbpcode to avoid the FileNotFoundError. But it is failing again for a reason I don't understand ^^' . Maybe it will work with a simple Retest this please ?

**Comment by matz-e** [comment]
Retest this please

I've cleaned out the PR directory, let's give this another try

**Comment by matz-e** [comment]
OK, finally noticed what's going on. The module generation fails, as upstream has added dependencies that are not present on your branch. If you rebase on `develop`, this should be successful.

**Review by matz-e — APPROVED** [review]
Hi, `git rebase bbp/develop` (or whatever you call our upstream) would have been cleaner to look at, but I think the squash-and-merge should take care of this.

In the future, you should be able to create branches in this repository, too. Then others can help out if quick fixes are needed.

**Comment by AurelienJaquier** [comment]
It worked! Thank you very much for your help!

**Pull request merged**